[Bug]: Extension error (sphinx.ext.linkcode) while building docs

### Bug summary

while building documentation the execution throws the below error and exit
```reading sources... [ 47%] api/backend_cairo_api
Extension error (sphinx.ext.linkcode):
Handler <function doctree_read at 0x0000026E71784700> for event 'doctree-read' threw an exception (exception: path is on mount 'C:', start on mount 'E:')
make: *** [Makefile:39: html] Error 2
```
[here](https://gist.github.com/sajjad-ali60/8de8842d19f64c7608a26f333c1d8098) is the log of `make html`

### Code for reproduction

followed the documentation for writing matplotlib docs


### Actual outcome

```
reading sources... [ 47%] api/backend_cairo_api
Extension error (sphinx.ext.linkcode):
Handler <function doctree_read at 0x0000026E71784700> for event 'doctree-read' threw an exception (exception: path is on mount 'C:', start on mount 'E:')
make: *** [Makefile:39: html] Error 2
```

### Expected outcome

it should print successfully build documentation and shows the html files in `build/html` folder

### Additional information

[requirements.txt](https://gist.github.com/sajjad-ali60/ced6a704d0d0cfdf4a8df668121d5e63) file
gcc version : 12.2.0
clang version : 14.0.6
VC++ compiler tools version : 14.34.31933

### Operating system

Windows 10 Pro Version	22H2

### Matplotlib Version

3.7.0.dev817+g9b7c902fd72e

### Matplotlib Backend

agg

### Python version

3.10.8

### Jupyter version

not installed

### Installation

from source (.tar.gz)

--- a/doc/conf.py
+++ b/doc/conf.py
@@ -682,7 +682,10 @@ if link_github:
                     if lineno else "")
 
         startdir = Path(matplotlib.__file__).parent.parent
-        fn = os.path.relpath(fn, start=startdir).replace(os.path.sep, '/')
+        try:
+            fn = os.path.relpath(fn, start=startdir).replace(os.path.sep, '/')
+        except ValueError:
+            return None
 
         if not fn.startswith(('matplotlib/', 'mpl_toolkits/')):
             return None

